### ISSUE TITLE
[compile error] Vcpkg & VTK failed on ${VTK_RENDERING_BACKEND_OPENGL_VERSION} macro

### ISSUE THREAD

**Issue by yoavmil** [PointCloudLibrary/pcl#5156]
TL;DR
using vcpkg, at [this line](https://github.com/PointCloudLibrary/pcl/blob/1ce45474b6e02d0561977f47d34a050e045cf812/pcl_config.h.in#L88)
`#define VTK_RENDERING_BACKEND_OPENGL_VERSION ${VTK_RENDERING_BACKEND_OPENGL_VERSION}`
the macro isn't defined nowhere, therefore it isn't a number, thus
```
#if VTK_RENDERING_BACKEND_OPENGL_VERSION < 2
  actor->GetMapper ()->ImmediateModeRenderingOn ();
#endif
```
[failes to compile](https://github.com/PointCloudLibrary/pcl/blob/1ce45474b6e02d0561977f47d34a050e045cf812/visualization/include/pcl/visualization/impl/pcl_visualizer.hpp#L637).

I'm using 
* MSVC 19
* VCPKG 2021-12-09
* pcl:x64-windows                                    1.12.0#2         Point Cloud Library (PCL) is open source library...
* pcl[visualization]:x64-windows                                      Build visualization
* vtk:x64-windows                                    9.0.3-pv5.9.1#5  Software system for 3D computer graphics, image ...

**Possible Solution**
Need someone that understands the vcpkg pipline

**Additional context**
This is the kinda thing that you can't fix cosmeticlty by just editing the file. It need to be solved in the root.
Also, after I have changed `#define VTK_RENDERING_BACKEND_OPENGL_VERSION 2` (set it to 2), other compilation and linkage errors happened.

**Comment by larshg** [comment]
@yoavmil is this still an issue? seems like you got some outputs, but maybe not visualized with PCL due to this error still?

**Comment by yoavmil** [comment]
Fixed at latest. Thanks.
